src/fcntl/statx.rs

```diff
@@ -0,0 +1,350 @@
+            None
+        }
+    }
+
+    /// Retrieve pointed-to device major and minor numbers (first and second
+    /// elements of the tuple respectively) of this character or block device
+    /// file, if it has been returned by kernel.
+    /// Note that this function does not check for the file type.
+    /// It would return `Some` even for regular files.
+    pub fn rdev_major_minor(&self) -> Option<(u32, u32)> {
+        // I'm not sure what exact mask bit should be used to check presence of this information.
+        // Actual Linux kernel seems return most of STATX_BASIC_STATS
+        // in one go, regarless of which things were asked for.
+        if Mask::STATX_BASIC_STATS.bits() & self.inner.stx_mask == Mask::STATX_BASIC_STATS.bits() {
+            Some((self.inner.stx_rdev_major, self.inner.stx_rdev_minor))
+        } else {
+            None
+        }
+    }
+
+    /// Determine if the file is compressed. None means kernel does not
+    /// indicate this attrbiute is supported by the filesystem
+    pub fn compressed(&self) -> Option<bool> {
+        if self.inner.stx_attributes_mask & libc::STATX_ATTR_COMPRESSED as u64 > 0 {
+            Some(self.inner.stx_attributes & libc::STATX_ATTR_COMPRESSED as u64 > 0)
+        } else {
+            None
+        }
+    }
+
+    /// Determine if the file is immutable. None means kernel does not indicate this
+    /// attrbiute is supported by the filesystem
+    pub fn immutable(&self) -> Option<bool> {
+        if self.inner.stx_attributes_mask & libc::STATX_ATTR_IMMUTABLE as u64 > 0 {
+            Some(self.inner.stx_attributes & libc::STATX_ATTR_IMMUTABLE as u64 > 0)
+        } else {
+            None
+        }
+    }
+    
+    /// Determine if the file is append-only. None means kernel does not indicate
+    /// this attrbiute is supported by the filesystem
+    pub fn append_only(&self) -> Option<bool> {
+        if self.inner.stx_attributes_mask & libc::STATX_ATTR_APPEND as u64 > 0 {
+            Some(self.inner.stx_attributes & libc::STATX_ATTR_APPEND as u64 > 0)
+        } else {
+            None
+        }
+    }
+
+    /// Determine if the file is not a candidate for a backup. None means kernel
+    /// does not indicate this attrbiute is supported by the filesystem
+    pub fn nodump(&self) -> Option<bool> {
+        if self.inner.stx_attributes_mask & libc::STATX_ATTR_NODUMP as u64 > 0 {
+            Some(self.inner.stx_attributes & libc::STATX_ATTR_NODUMP as u64 > 0)
+        } else {
+            None
+        }
+    }
+
+    /// Determine if the file requires a key to be encrypted(?).
+    /// None means kernel does not indicate this attrbiute is supported by the filesystem
+    pub fn encrypted(&self) -> Option<bool> {
+        if self.inner.stx_attributes_mask & libc::STATX_ATTR_ENCRYPTED as u64 > 0 {
+            Some(self.inner.stx_attributes & libc::STATX_ATTR_ENCRYPTED as u64 > 0)
+        } else {
+            None
+        }
+    }
+
+    /*
```

Comment:
Why are these two methods commented out?

Reply:
Because of current version of `libc` crate does not expose e.g. `STATX_ATTR_VERITY`. See my question at https://github.com/nix-rust/nix/issues/1649#issuecomment-1033938084.
Android target's libc lacks even more consts in libc.

Reply:
I submitted a [PR](https://github.com/rust-lang/libc/pull/2941) to add the missing constants on GNU/Linux